Assets/Scripts/UI/Overlay/OverlayMap.cs

```diff
@@ -271,7 +271,14 @@ public void SetOverlay(string name)
 
                 Tile tile = WorldController.Instance.GetTileAtWorldCoord(new Vector3(x, y, z));
                 DynValue result = FunctionsManager.Overlay.Call(descr.LuaFunctionName, new object[] { tile, World.Current });
-                return (int)result.ToScalar().CastToNumber();
+                double? value = result.CastToNumber();
+                if (value == null)
+                {
+                    Debug.ULogErrorChannel("OverlayMap", string.Format("The return value from the function named '{0}' was null", descr.LuaFunctionName));
```

Comment:
I guess it makes sense since this callback is called once per tile. If we go back to what it was before. I can make it log once.